manager/controlapi/task.go

```diff
@@ -104,10 +106,18 @@ func (s *Server) ListTasks(ctx context.Context, request *api.ListTasksRequest) (
 	if request.Filters != nil {
 		tasks = filterTasks(tasks,
 			func(e *api.Task) bool {
-				return filterContains(e.ServiceAnnotations.Name, request.Filters.Names)
+				name := e.Annotations.Name
```

Comment:
@aaronlehmann And my answer was that we need to index one or the other, depending on the task data.